openassessment/xblock/validation.py

```diff
@@ -47,9 +47,13 @@ def _duplicates(items):
 
 def _is_valid_assessment_sequence(assessments):
     """
-    Check whether the sequence of assessments is valid.
-    For example, we currently allow self-assessment after peer-assessment,
-    but do not allow peer-assessment before self-assessment.
+    Check whether the sequence of assessments is valid. The rules enforced are:
+        -must have one of staff-, peer-, self-, or example-based-assessment listed
+        -in addition to those, only student-training is a valid entry
+        -no duplicate entries
+        -if staff-assessment is present, it must come last
```

Comment:
You're correct, the grading can take place at any time.

This order will make staff grading appear last on the LMS problem visually, but will not affect the ability of staff tools to grab the assessment and grade it at any point in the process.